### PR TITLE
tikv: switch follower read request to other follower when meet region_not_found

### LINKED ISSUE #17454: region unavailable error when region split but some followers are isolated
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

isolate some tikv, trigger split region, and access a new region

### 2. What did you expect to see? (Required)

success

### 3. What did you see instead (Required)

failed with "region unavailable"

### 4. Affected version (Required)

4.0

### 5. Root Cause Analysis

new region's follower didn't finish apply and TiDB access it without retry, it's better retry other peers or failback to leader reader when meet region_not_found

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve?

Issue Number: close #17454 

Problem Summary:

after region split, some follower didn't finish apply due to some reason (i.e. the follower be isolated)

it's better to retry other peers.

### What is changed and how it works?

What's Changed:

increase seed to let retry request to other peers.

### Related changes

- Need to cherry-pick to the release branch 4.0

### Check List 

Tests 

- Unit test

Side effects

- n/a

### Release note 

- Fix Region unavailable error when region split but some followers are isolated

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/17453)

### CODE REVIEW

File: store/tikv/region_request.go [hunk 1]
```diff
@@ -408,6 +411,11 @@ func (s *RegionRequestSender) onRegionError(bo *Backoffer, ctx *RPCContext, seed
 		logutil.BgLogger().Warn("tikv reports `RaftEntryTooLarge`", zap.Stringer("ctx", ctx))
 		return false, errors.New(regionErr.String())
 	}
+	if regionErr.GetRegionNotFound() != nil && seed != nil {
+		logutil.BgLogger().Debug("tikv reports `RegionNotFound` in follow-reader",
```
Comment (crazycs520) [inline store/tikv/region_request.go:415]: Sorry to bother, How to make sure it is a `follower-read` here? 
need to check `req.ReplicaRead == true` to make sure it is a `follower-read`?

Comment (lysu) [reply store/tikv/region_request.go:415]: :D we can make sure it by see if `seed == nil`， seed only be nil when it's follower read？

File: store/tikv/region_request.go [hunk 2]
```diff
@@ -408,6 +411,11 @@ func (s *RegionRequestSender) onRegionError(bo *Backoffer, ctx *RPCContext, seed
...
+			zap.Stringer("ctx", ctx), zap.Uint32("seed", *seed))
+		*seed = *seed + 1
```
Comment (lzmhhh123) [inline store/tikv/region_request.go:417]: Can we be certain that if the seed is changed, then we can switch to other followers?

Comment (lzmhhh123) [reply store/tikv/region_request.go:417]: Just curious about this.

Comment (lysu) [reply store/tikv/region_request.go:417]: maybe we'd better remove seed mechanism later Orz..

**Pull request merged**